packages/app/src/components/DeploymentStatusCard/DeploymentStatusCard.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+  Progress,
+  StatusAborted,
+  StatusError,
+  StatusOK,
+  StatusPending,
+  StatusWarning,
+} from '@backstage/core-components';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import {
+  detectErrors,
+  useKubernetesObjects,
+} from '@backstage/plugin-kubernetes';
+import { Typography } from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import StatusCard from '../catalog/shared/StatusCard';
+
+type Phase = {
+  running: number;
+  pending: number;
+  succeeded: number;
+  failed: number;
+  unknown: number;
+  hasError: number;
+};
+
+const Status = ({ pods }: { pods: Phase }) => {
+  if (pods.failed) {
+    return <StatusError />;
+  }
+  if (pods.hasError) {
+    return <StatusWarning />;
+  }
+  if (pods.pending) {
+    return <StatusPending />;
+  }
+  if (pods.running) {
+    return <StatusOK />;
+  }
+  return <StatusAborted />;
+};
+
+const DeploymentStatusCard = () => {
+  const { entity } = useEntity();
+  const [pods, setPods] = useState({} as Phase);
+
+  const { kubernetesObjects, error } = useKubernetesObjects(entity, 30000);
+
+  useEffect(() => {
+    if (kubernetesObjects && kubernetesObjects?.items.length > 0) {
```

Comment:
Can this if be reverted so that it exits the `useEffect()` function when the objects don't exist?